hidimstat/loco.py

```diff
@@ -0,0 +1,183 @@
+import numpy as np
+from joblib import Parallel, delayed
+from sklearn.base import (
+    BaseEstimator,
+    TransformerMixin,
+    check_is_fitted,
+    clone,
+)
+from sklearn.metrics import mean_squared_error
+
+
+class LOCO(BaseEstimator, TransformerMixin):
+    """
+    Leave-One-Covariate-Out (LOCO) algorithm.
+
+
+    Parameters
+    ----------
+    estimator: scikit-learn compatible estimator
+        The predictive model.
+    groups: dict, default=None
+        Dictionary of groups for the covariates. The keys are the group names
+        and the values are lists of covariate indices.
+    loss: callable, default=mean_squared_error
+        Loss function to evaluate the model performance.
+    score_proba: bool, default=False
+        Whether to use the predict_proba method of the estimator.
+    random_state: int, default=None
+        Random seed for the permutation.
+    n_jobs: int, default=1
+        Number of jobs to run in parallel.
+
+    References
+    ----------
+    .. footbibliography::
```

Comment:
Add a reference to clarify which version of LOCO is used here

Reply:
The current implementation uses the LOCO described by @achamma723's paper (now cited).